07/AcyclicGraph.scala

```diff
@@ -0,0 +1,214 @@
+import scala.io.Source
+
+object AcyclicGraph {
+case class Worker(step: Step, startTick: Int)
```

Comment:
Instead of `startTick` you might prefer to use `finishTick` or `firstFreeTick`. This could simplify some of the functions.